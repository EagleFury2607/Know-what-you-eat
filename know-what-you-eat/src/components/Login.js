import React,{ Component } from 'react';
import axios from 'axios';

class Login extends Component{
    constructor(props){
        super(props);
        this.headers = {
            'Content-Type':'application/json;charset=UTF-8'
        }
        this.state = {
            username: '',
            password: ''
        }
        this.change = this.change.bind(this);
        this.submit = this.submit.bind(this);
    }
    change(e){
        this.setState({
            [e.target.name]:e.target.value
        })
    }
    
    submit(e){
        e.preventDefault();
        axios.post('/login',{
            username:this.state.username,
            password:this.state.password
        },this.headers).then((result) => {
            localStorage.setItem('jwt',result.data);
            console.log(result.data);
        });
    }
    render(){
        return(<div>
            <form onSubmit={e => this.submit(e)}>
                <label>Email</label><input type="text" name="username" onChange={e => this.change(e)} value={this.state.username}/>
                <label>Password</label><input type="password" name="password" onChange={e => this.change(e)} 
                value={this.state.password}/>
                <button type="submit" >Submit</button>
            </form>
        </div>
        );
        
    }
}

export default Login;