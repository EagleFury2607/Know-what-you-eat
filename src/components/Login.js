import React, { Component } from "react";
import axios from "axios";
import "./css/Login.css";
import NavbarComponent from "./NavbarComponent";

class Login extends Component {
  constructor(props) {
    super(props);
    this.headers = {
      "Content-Type": "application/json;charset=UTF-8",
    };
    this.state = {
      username: "",
      password: "",
    };
    this.change = this.change.bind(this);
    this.submit = this.submit.bind(this);
  }
  change(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  submit(e) {
    e.preventDefault();
    axios
      .post(
        "http://test.beserver.cloudns.cl/login",
        {
          username: this.state.username,
          password: this.state.password,
        },
        this.headers
      )
      .then((result) => {
        if (result.status === 200) {
          if (result.data.status === 301) {
            alert(result.data.response);
          } else if (result.data.status === 200) {
            localStorage.setItem("jwt", result.data.token);
            this.props.history.push("/Dashboard");
          }
        }
      })
      .catch((error) => {
        if (!error.status) {
          alert("Server is down !");
        }
      });
  }
  render() {
    return (
      <div>
        <NavbarComponent userLogginStatus={"Register"} />
        <div id="login-background">
          <form
            id="custom-form"
            className="text-center Login.loginClass form-signin"
            onSubmit={(e) => this.submit(e)}
          >
            <h1 className="h3 mb-3 font-weight-normal">Please sign in</h1>
            <input
              className="form-control"
              placeholder="Email address"
              type="email"
              name="username"
              onChange={(e) => this.change(e)}
              value={this.state.username}
              required
              autoFocus
            />
            <input
              className="form-control"
              type="password"
              name="password"
              onChange={(e) => this.change(e)}
              value={this.state.password}
              required
              placeholder="password"
            />
            <button type="submit" className="btn btn-lg btn-primary btn-block">
              Submit
            </button>
          </form>
        </div>
      </div>
    );
  }
}

export default Login;
