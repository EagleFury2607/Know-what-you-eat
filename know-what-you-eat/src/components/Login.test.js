import React from "react";
import { shallow, mount } from "enzyme";
import Login from "./Login";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe("Login Component Unit Test",()=>{
	test("Login component loads", () => {
		const wrapper = shallow(<Login />);
		expect(wrapper).toMatchSnapshot();
	  });
	test("Contains form element", () => {
		const wrapper = shallow(<Login />);
		expect(false);
	  });
});
