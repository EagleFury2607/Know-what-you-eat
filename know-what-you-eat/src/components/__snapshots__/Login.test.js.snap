exports[`Login Component Unit Test Login component loads 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(NavbarComponent)
          userLogginStatus="Register" />,
        <form
          className="text-center Login.loginClass form-signin"
          id="custom-form"
          onSubmit={[Function]}>
          <h1
                    className="h3 mb-3 font-weight-normal">
                    Please sign in
          </h1>
          <input
                    autoFocus={true}
                    className="form-control"
                    name="username"
                    onChange={[Function]}
                    placeholder="Email address"
                    required={true}
                    type="email"
                    value="" />
          <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    required={true}
                    type="password"
                    value="" />
          <button
                    className="btn btn-lg btn-primary btn-block"
                    type="submit">
                    Submit
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "userLogginStatus": "Register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="h3 mb-3 font-weight-normal">
              Please sign in
</h1>,
            <input
              autoFocus={true}
              className="form-control"
              name="username"
              onChange={[Function]}
              placeholder="Email address"
              required={true}
              type="email"
              value="" />,
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              placeholder="password"
              required={true}
              type="password"
              value="" />,
            <button
              className="btn btn-lg btn-primary btn-block"
              type="submit">
              Submit
</button>,
          ],
          "className": "text-center Login.loginClass form-signin",
          "id": "custom-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please sign in",
              "className": "h3 mb-3 font-weight-normal",
            },
            "ref": null,
            "rendered": "Please sign in",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "className": "form-control",
              "name": "username",
              "onChange": [Function],
              "placeholder": "Email address",
              "required": true,
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "required": true,
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn btn-lg btn-primary btn-block",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(NavbarComponent)
            userLogginStatus="Register" />,
          <form
            className="text-center Login.loginClass form-signin"
            id="custom-form"
            onSubmit={[Function]}>
            <h1
                        className="h3 mb-3 font-weight-normal">
                        Please sign in
            </h1>
            <input
                        autoFocus={true}
                        className="form-control"
                        name="username"
                        onChange={[Function]}
                        placeholder="Email address"
                        required={true}
                        type="email"
                        value="" />
            <input
                        className="form-control"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        required={true}
                        type="password"
                        value="" />
            <button
                        className="btn btn-lg btn-primary btn-block"
                        type="submit">
                        Submit
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "userLogginStatus": "Register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="h3 mb-3 font-weight-normal">
                Please sign in
</h1>,
              <input
                autoFocus={true}
                className="form-control"
                name="username"
                onChange={[Function]}
                placeholder="Email address"
                required={true}
                type="email"
                value="" />,
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="password"
                required={true}
                type="password"
                value="" />,
              <button
                className="btn btn-lg btn-primary btn-block"
                type="submit">
                Submit
</button>,
            ],
            "className": "text-center Login.loginClass form-signin",
            "id": "custom-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please sign in",
                "className": "h3 mb-3 font-weight-normal",
              },
              "ref": null,
              "rendered": "Please sign in",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "className": "form-control",
                "name": "username",
                "onChange": [Function],
                "placeholder": "Email address",
                "required": true,
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "required": true,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn btn-lg btn-primary btn-block",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
